base/src/test/proto/spine/test/validate/msg/validating_builder_test.proto

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright 2017, TeamDev Ltd. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+syntax = "proto3";
+
+package spine.test.validate.msg;
+
+option (type_url_prefix) = "type.spine.io";
+option java_generate_equals_and_hash = true;
+option java_multiple_files = true;
+option java_outer_classname = "ValidatingBuilderShouldProto";
+option java_package = "io.spine.test.validate.msg.builder";
+
+import "google/protobuf/timestamp.proto";
+
+import "spine/options.proto";
+
+import "spine/net/email_address.proto";
+import "spine/people/person_name.proto";
+
+message Customer {
```

Comment:
Judging on its contents, this isn't a `Customer`. All the repeated fields look weird in this context. Please rethink the example. I would use `Project` with attachments, tasks, members. Some of those could have nested collections; e.g. a task should probably have attachments (as well as a project), comments etc. 

`attachment` can be a map (MIME type string -> `Attachment`). And if you need another map, you could have something like `attributes`, like we have for a `Command` message.